### PR TITLE
Sync main with deepmind

### PR DESCRIPTION
**Pull request by thowell**

### CODE REVIEW

File: grpc/agent_service_test.cc [hunk 1]
```diff
@@ -31,6 +32,7 @@
 #include <mujoco/mujoco.h>
 #include "grpc/agent.grpc.pb.h"
 #include "grpc/agent.pb.h"
+#include "third_party/mujoco_mpc/grpc/agent.proto.h"
```
Comment (erez-tom) [inline grpc/agent_service_test.cc:35]: this is wrong.  I'll go fix the underlying cause, but can you please fix this here?

File: grpc/grpc_agent_util.cc [hunk 2]
```diff
@@ -142,39 +142,71 @@ grpc::Status SetState(const SetStateRequest* request, mjpc::Agent* agent,
...
                        const mjpc::Agent* agent,
                        const mjModel* model, mjData* rollout_data,
                        mjpc::State* rollout_state,
                        GetActionResponse* response) {
-  int nu = agent->GetActionDim();
-  std::vector<double> ret = std::vector<double>(nu, 0);
-
   double time =
-      request->has_time() ? request->time() : agent->state.time();
+      request->has_time() ? request->time() : agent->ActiveState().time();
```
Comment (erez-tom) [inline grpc/grpc_agent_util.cc:187]: This is a regression.

File: grpc/grpc_agent_util.cc [hunk 3]
```diff
@@ -142,39 +142,71 @@ grpc::Status SetState(const SetStateRequest* request, mjpc::Agent* agent,
...
-    agent->state.CopyTo(model, rollout_data);
-    rollout_data->time = time;
-    double end_time = time + request->averaging_duration();
-    int nactions = 0;
-    while (rollout_data->time <= end_time) {
+    if (request->nominal_action()) {
+      rollout_data = nullptr;
+      rollout_state = nullptr;
+    } else {
+      agent->ActiveState().CopyTo(model, rollout_data);
```
Comment (erez-tom) [inline grpc/grpc_agent_util.cc:194]: This needs updating.

File: grpc/grpc_agent_util.cc [hunk 4]
```diff
@@ -142,39 +142,71 @@ grpc::Status SetState(const SetStateRequest* request, mjpc::Agent* agent,
...
+                        request->nominal_action(), rollout_data, rollout_state,
+                        time, request->averaging_duration());
+    response->mutable_action()->Assign(ret.begin(), ret.end());
   } else {
-    agent->ActivePlanner().ActionFromPolicy(
-        ret.data(), &agent->state.state()[0], time);
+    std::vector<double> ret(model->nu, 0);
+    const double* state = request->nominal_action()
+                              ? nullptr
+                              : agent->ActiveState().state().data();
```
Comment (erez-tom) [inline grpc/grpc_agent_util.cc:205]: Also needs updating.

File: grpc/agent_service_test.cc (hunk 1 above)
Comment (erez-tom) [reply grpc/agent_service_test.cc:35]: I fixed it in https://github.com/deepmind/mujoco_mpc/commit/7c9c7a6c7766042a9107a95f07d7d09a5212a0b9
Please update the PR and check that it's fixed.

Comment (erez-tom) [reply grpc/agent_service_test.cc:35]: Can you update the PR and make sure it goes away?  Supposedly I fixed it in https://github.com/deepmind/mujoco_mpc/commit/7c9c7a6c7766042a9107a95f07d7d09a5212a0b9

Comment (thowell) [reply grpc/agent_service_test.cc]: Fixed. 

There is a similar "third_party" issue in interface.cc that I fixed too.

File: grpc/grpc_agent_util.cc (hunk 2 above)
Comment (thowell) [reply grpc/grpc_agent_util.cc]: Fixed.

File: grpc/grpc_agent_util.cc (hunk 3 above)
Comment (thowell) [reply grpc/grpc_agent_util.cc]: Fixed.

File: grpc/grpc_agent_util.cc (hunk 4 above)
Comment (thowell) [reply grpc/grpc_agent_util.cc]: Fixed.

**Comment by thowell** [comment]
Fixed one small spelling error in the README.

**Review by erez-tom — APPROVED** [review]

**Pull request merged**